modules/masterbar/masterbar.php

```diff
@@ -68,6 +76,104 @@ function __construct() {
+			'wp-admin-bar-discover-recommended-blogs' => array( 'masterbar_reader_recommendations_link_click', 'https://wordpress.com/recommendations' ),
+			'wp-admin-bar-my-activity-my-likes'       => array( 'masterbar_reader_my_links_link_click', 'https://wordpress.com/activities/likes' ),
+
+			//account
+			'wp-admin-bar-user-info'                  => array( 'masterbar_user_name_link_click', 'https://wordpress.com' ),
+			// account - profile
+			'wp-admin-bar-my-profile'                 => array( 'masterbar_profile_my_profile_link_click', 'https://wordpress.com/me' ),
+			'wp-admin-bar-account-settings'           => array( 'masterbar_profile_account_settings_link_click', 'https://wordpress.com/me/account' ),
+			'wp-admin-bar-billing'                    => array( 'masterbar_profile_manage_purchases_link_click', 'https://wordpress.com/me/purchases' ),
+			'wp-admin-bar-security'                   => array( 'masterbar_profile_security_link_click', 'https://wordpress.com/me/security' ),
+			'wp-admin-bar-notifications'              => array( 'masterbar_profile_notifications_link_click', 'https://wordpress.com/me/notifications' ),
+			//account - special
+			'wp-admin-bar-get-apps'                   => array( 'masterbar_get_apps_link_click', 'https://wordpress.com/me/get-apps' ),
+			'wp-admin-bar-next-steps'                 => array( 'masterbar_next_steps_link_click', 'https://wordpress.com/me/next' ),
+			'wp-admin-bar-help'                       => array( 'masterbar_help_link_click', 'https://jetpack.com/support/' ),
+		);
+	}
+
+	function tracks_and_redirect_allow_wpcom_domain( $domains ) {
+		if ( empty( $domains ) ) {
+			$domains = array();
+		}
+		$domains[] = 'wordpress.com';
+		return array_unique( $domains );
+	}
+
+	public function tracks_and_redirect_add_route() {
+		add_rewrite_rule( 'jetpack-track-and-bounce.php\?tracks_and_bounce=(\w*)', 'index.php?tracks_and_bounce=$matches[1]', 'top' );
+	}
+
+	public function tracks_and_redirect_add_query_vars( $query_vars ) {
+		$query_vars[] = 'tracks_and_bounce';
+		$query_vars[] = 'tracks_and_bounce_nonce';
+		return $query_vars;
+	}
+
+	public function tracks_and_redirect_parse_request( $query ) {
+		if ( ! array_key_exists( 'tracks_and_bounce', $query->query_vars ) ) {
+			return;
+		}
+
+		if ( ! wp_verify_nonce( $query->query_vars[ 'tracks_and_bounce_nonce' ], 'jp-tracks-masterbar-nonce' ) ) {
+			// no nonce, wrong link or missing tracking information, push back to settings page
+			wp_safe_redirect(
+				add_query_arg(
+					array( 'page' => 'jetpack-settings' ),
+					network_admin_url( 'admin.php' )
+				)
+			);
+			die();
+		}
+
+		$trackId = $query->query_vars[ 'tracks_and_bounce' ];
+		if ( ! array_key_exists( $trackId, $this->tracks_events ) ) {
+			return;
+		}
+
+		$track_event = $this->tracks_events[ $trackId ];
+
+		JetpackTracking::record_user_event(
+			$track_event[0],
+			array( 'target' => $track_event[1] )
```

Comment:
What's the benefit of recording `target` here? I don't expect these to change, and even if they do it seems to me that we are only interested in recording user's action of navigating to particular page, and not in the corresponding URL.

Reply:
Having a single tracks event now, it make sense to store the target.